src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseType.java

```diff
@@ -153,12 +154,34 @@ public void setSearchAliasFields(List<SearchAliasField> searchAliasFields) {
     }
 
     @JsonIgnore
-    public Optional<CaseField> getCaseField(String caseFieldId) {
-        return caseFields.stream().filter(caseField -> caseField.getId().equalsIgnoreCase(caseFieldId)).findFirst();
+    public boolean isCaseFieldACollection(String caseFieldId) {
+        return getCaseField(caseFieldId).map(CaseField::isCollectionFieldType).orElse(false);
     }
 
+    /**
+     * Gets a caseField by caseField id and specified path.
+     *
+     * @param caseFieldId CaseField id
+     * @param path Path to a nested CaseField
+     * @return Optional of a CaseField
+     */
     @JsonIgnore
-    public boolean isCaseFieldACollection(String caseFieldId) {
-        return getCaseField(caseFieldId).map(CaseField::isCollectionFieldType).orElse(false);
+    public Optional<CaseField> getCaseFieldByPath(String caseFieldId, String path) {
```

Comment:
I think you can simplify the method to:
```
Optional<CaseField> caseFieldOptional = getCaseField(caseFieldId);

        if (StringUtils.isBlank(path)) {
            return caseFieldOptional;
        }
        return caseFieldOptional.map(f-> f.findNestedElementByPath(path));
```

Reply:
I'm just realising now that the path is relative to a field id, it's not an absolute path. In this case you were right this method belongs to the CaseField not the CaseType